marine/marine.py

```diff
@@ -154,3 +175,37 @@ def _add_or_get_filter(
 
     def __del__(self):
         self._marine.destroy_marine()
+
+    @staticmethod
+    def replace_macros(
+        fields: List[str], macros: Optional[Dict[str, List[str]]]
+    ) -> Tuple[List[str], List[str]]:
+        demacroed_fields = []
+        used_macros = []
+
+        if not macros:
+            return fields, used_macros
+
+        for field in fields:
+            if field in macros.keys():
+                used_macros.append(field)
+                demacroed_fields.extend(macros[field])
+            else:
+                demacroed_fields.append(field)
+
+        return demacroed_fields, used_macros
+
+    @staticmethod
+    def restore_macros(
+        result: Dict[str, str], macros: Optional[Dict[str, List[str]]], used_macros: List[str]
+    ) -> Dict[str, str]:
+        for macro in used_macros:
+            macro_value = ""
```

Comment:
Pretty sure this function (and therfore `filter_and_parse`) won't work when I have the same field appearing in several macros. Verify that, test that and rewrite.
Plus this function is rather awkward and should be rewritten regardless.
Consult @royreznik or me if needed :)